src/vendor/github.com/xelabs/go-mysqlstack/sqlparser/sql.y

```diff
@@ -2542,66 +2571,167 @@ column_comment_opt:
 		$$ = &IndexDefinition {
-			Type: string($1) + " " + string($2),
-			Name: NewColIdent("PRIMARY"),
-			Opts: NewIndexOptions($4, nil),
-			Primary: true,
-			Unique: true,
+			Type: string($1),
+			Name: NewColIdent($2),
+			Opts: NewIndexOptions($5, $7),
+			Primary: false,
+			Unique: false,
 		}
 	}
-|	UNIQUE index_or_key ID '(' index_column_list ')'  normal_key_opts
+|   spatial_or_fulltext index_or_key_opt index_name '(' index_column_list ')' index_options
 	{
+        typ := string($1)
+        if $3 != "" {
+            typ = typ + " " + string($2)
+        }
 		$$ = &IndexDefinition {
-			Type: string($1) + " " + string($2),
-			Name: NewColIdent(string($3)),
+			Type: typ,
+			Name: NewColIdent($3),
 			Opts: NewIndexOptions($5, $7),
 			Primary: false,
-			Unique: true,
+			Unique: false,
 		}
 	}
-|	UNIQUE ID '(' index_column_list ')' normal_key_opts
+|   constraint_keyword_opt PRIMARY KEY index_type_opt '(' index_column_list ')' index_options
 	{
+        // TODO(): in the future we'll support format out index_type, currently skip it
 		$$ = &IndexDefinition {
-			Type: string($1),
-			Name: NewColIdent(string($2)),
-			Opts: NewIndexOptions($4, $6),
-			Primary: false,
+			Type: string($2) + " " + string($3),
+			Name: NewColIdent("PRIMARY"),
+			Opts: NewIndexOptions($6, $8),
+			Primary: true,
 			Unique: true,
 		}
 	}
-|	index_or_key ID '(' index_column_list ')' normal_key_opts
+|	constraint_keyword_opt UNIQUE index_or_key_opt index_name index_type_opt '(' index_column_list ')'  index_options
 	{
+        // TODO(): in the future we'll support format out index_type, currently skip it
+        typ := string($2)
+        if $3 != "" {
+            typ = typ + " " + string($3)
+        }
 		$$ = &IndexDefinition {
-			Type: string($1),
-			Name: NewColIdent(string($2)),
-			Opts: NewIndexOptions($4, $6),
+			Type: typ,
+			Name: NewColIdent($4),
+			Opts: NewIndexOptions($7, $9),
 			Primary: false,
-			Unique: false,
+			Unique: true,
 		}
 	}
-|	FULLTEXT index_or_key ID '(' index_column_list ')' fulltext_key_opts
+
+index_name:
+    {
+        $$ = ""
+    }
+|   ID
+    {
+        $$ = string($1)
+    }
+
+index_option:
+	KEY_BLOCK_SIZE opt_equal INTEGRAL
 	{
-		$$ = &IndexDefinition {
-			Type: string($1) + " " + string($2),
-			Name: NewColIdent(string($3)),
-			Opts: NewIndexOptions($5, $7),
-			Primary: false,
-			Unique: false,
+		$$ = &IndexOption{
+			Type: IndexOptionBlockSize,
+			Val:  NewIntVal($3),
 		}
 	}
-|	SPATIAL index_or_key ID '(' index_column_list ')' spatial_key_opts
+|	index_type
+    {
+		$$ = &IndexOption{
+			Type: IndexOptionUsing,
+			Val:  NewStrValWithoutQuote($1),
+		}
+    }
+|	COMMENT_KEYWORD STRING
 	{
-		$$ = &IndexDefinition {
-			Type: string($1) + " " + string($2),
-			Name: NewColIdent(string($3)),
-			Opts: NewIndexOptions($5, $7),
-			Primary: false,
-			Unique: false,
+		$$ = &IndexOption{
+			Type: IndexOptionComment,
+			Val:  NewStrVal($2),
 		}
 	}
+|	WITH PARSER NGRAM
```

Comment:
There is not only one parser `NGRAM`. Now that it has been modified, the restrictions can be relaxed here. Maybe replace with `parser_name` such as `ID`